datasets/eu/esma_saris/crawler.py

```diff
@@ -0,0 +1,71 @@
+from zavod import Context, helpers
+import csv
+
+
+def crawl(context: Context) -> None:
+    # Set the initial cookie
+    data = {
+        "core": "esma_registers_saris_new",
+        "pagingSize": "10",
+        "start": 0,
+        "keyword": "",
+        "sortField": "effectiveFrom asc",
+        "criteria": [],
+        "wt": "json",
+    }
+    context.http.post(
+        "https://registers.esma.europa.eu/publication/searchRegister/doMainSearch",
+        json=data,
+    )
+    source_file = context.fetch_resource(
+        "source.csv",
+        context.dataset.data.url,
+    )
+    with open(source_file, "r") as f:
+        reader = csv.DictReader(f)
+        for row in reader:
+            entity = context.make("Security")
+            isin = row.pop("instrumentIdentifier")
+            if isin is None:
+                context.log.warn("No ISIN", row=row)
+                return
+            entity.id = f"isin-{isin}"
+            entity.add("isin", isin)
+            entity.add("name", row.pop("instrumentFullName", isin))
+
+            sanction = helpers.make_sanction(context, entity)
+            sanction.id = context.make_id(row.pop("id"))
+            sanction.add("program", "ESMA")
```

Comment:
I don't think we need this: the sanction['authority'] is already going to contain the `publisher.name`.

Reply:
Yeah, this is a bit of a weird one. As far as I can understand ([basis here](https://www.esma.europa.eu/publications-and-data/interactive-single-rulebook/mifid-ii/article-52-suspension-and-removal)) ESMA is sort of an "aggregator" of suspensions and removals.

Is the program the name of directive itself maybe? Or do we leave it blank